src/store/actions.js

```diff
@@ -767,18 +775,20 @@ export default {
 			}
 
 			const ids = getters.getEnvelopes(mailboxId, query).map((env) => env.databaseId)
-			logger.debug(`mailbox sync of ${mailboxId} (${query}) has ${ids.length} known IDs`)
-			return syncEnvelopes(mailbox.accountId, mailboxId, ids, query, init)
+			const lastTimestamp = getters.getPreference('sort-order') === 'newest' ? null : getters.getEnvelopes(mailboxId, query)[0].dateInt
```

Comment:
```suggestion
			const lastTimestamp = getters.getPreference('sort-order') === 'newest' ? null : getters.getEnvelopes(mailboxId, query)[0]?.dateInt
```

fixes https://github.com/nextcloud/mail/pull/8231#issuecomment-1800110888